### PR TITLE
PR IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator: Initial release + DRG UnitTest automation Fix

### PR DESCRIPTION
**Pull request by William-Francillette**
#### Pull Request (PR) description

Resource for Intune policy: 
Android Device Administrator device restriction policies

+ Fixed UnitTest automation and added support for CIM instances in test

### CODE REVIEW

**Comment by andikrueger** [comment]
Could you update the change log with the changes you made?

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator/MSFT_IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator.psm1 [hunk 1]
```diff
@@ -0,0 +1,2167 @@
...
+        if($null -eq $context)
+        {
+            $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftGraph' `
+                -InboundParameters $PSBoundParameters -ProfileName 'beta'
+        }
+        Select-MgProfile 'beta' -ErrorAction Stop
+    }
+    catch
+    {
+        Write-Verbose -Message "Reloading1"
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator/MSFT_IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator.psm1:272]: Can we make sure to remove this testing verbose?

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator/MSFT_IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator.schema.mof [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    [Write, Description("Kiosk mode managed app id")] String appId;
+    [Write, Description("")] String appStoreUrl;
+    [Write, Description("")] String name;
+    [Write, Description("")] String publisher;
+};
+
+[ClassVersion("1.0.0.0"), FriendlyName("IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator")]
+class MSFT_IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator : OMI_BaseResource
+{
+    [Write, Description("")] String Id;
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator/MSFT_IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator.schema.mof:23]: Please provide description for the 3 parameters

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator/MSFT_IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator.schema.mof [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    [Write, Description("Number of consecutive times an incorrect password can be entered before device is wiped of all data.")] UInt32 PasswordSignInFailureCountBeforeFactoryReset;
+    [Write, Description("Block user from powering off device. If this setting is disabled the setting 'Number of sign-in failures before wiping device' does not function.")] Boolean PowerOffBlocked;
+    [Write, Description(""), ValueMap{"none","low","medium","high"}, Values{"none","low","medium","high"}] String RequiredPasswordComplexity;
+    [Write, Description("Block capturing contents of screen as an image.")] Boolean ScreenCaptureBlocked;
+    [Write, Description("Security require verify apps")] Boolean SecurityRequireVerifyApps;
+    [Write, Description("Block sync with Google backup.")] Boolean StorageBlockGoogleBackup;
+    [Write, Description("Block removable storage usage (Samsung KNOX Standard 4.0+).")] Boolean StorageBlockRemovableStorage;
+    [Write, Description("Require encryption on device. Not all devices support encryption.<br>
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator/MSFT_IntuneDeviceConfigurationPolicyAndroidDeviceAdministrator.schema.mof:66]: Can we please make the description on one line and remove the HTML tags even if that may make it harder to read? Thanks

**Review by NikCharlebois — CHANGES_REQUESTED** [review]

**Comment by William-Francillette** [comment]
@NikCharlebois  Same as IntuneDeviceConfigurationPolicyAndroidOpenSourceProject
Unit test pass in local
image

But fails in dev due to O365User
image

**Pull request merged**